relational_operators/CreateIndexOperator.hpp

```diff
@@ -75,7 +83,8 @@ class CreateIndexOperator : public RelationalOperator {
 
  private:
   CatalogRelation *relation_;
-  const std::string &index_name_;
+  std::string index_name_;
```

Comment:
We avoid deep copies as possible for the sake of high performance.

The original code should be ok in the distributed case, and I think it is still good for concurrent queries.

Under which scenario do you think the referenced string will change? It should be fixed during the query optimization, right?